source/Calamari.Common/Plumbing/FileSystem/ICalamariFileSystem.cs

```diff
@@ -23,9 +23,13 @@ public interface ICalamariFileSystem
         long GetFileSize(string path);
         string ReadFile(string path);
         string ReadFile(string path, out Encoding encoding);
-        void OverwriteFile(string path, string? contents);
-        void OverwriteFile(string path, string? contents, Encoding encoding);
+        void OverwriteFile(string path, string contents, Encoding? encoding = null);
+        void OverwriteFile(string path, Func<string> getText, Encoding? encoding = null);
+        void OverwriteFile(string path, Action<TextWriter> writeToWriter, Encoding? encoding = null);
         void OverwriteFile(string path, byte[] data);
+        void WriteAllText(string path, string contents, Encoding? encoding = null);
+        void WriteAllText(string path, Func<string> getText, Encoding? encoding = null);
```

Comment:
What's the use case for this overload? Is it for cases where generating the text is expensive so you want to do it lazily in case some precondition fails?

Reply:
You're right, that's become redundant. I'll trim it.

Reply:
That's done.